src/stm32/fdcan.c

```diff
@@ -251,6 +274,29 @@ can_init(void)
 
     SOC_CAN->NBTP = btr;
 
+#if CONFIG_MACH_STM32H7
+    /*
+        The Message RAM of STM32H7 is settable
+        So we set it to be consistent with STM32G0
+     */
+    uint32_t flssa = (uint32_t)&MSG_RAM - (uint32_t)&fdcan_ram->fdcan1;
+    uint32_t f0sa = flssa +
+                    (((uint32_t)MSG_RAM.RXF0 - (uint32_t)MSG_RAM.FLS) / 4);
+    uint32_t tbsa = f0sa +
+                    (((uint32_t)MSG_RAM.TXFIFO - (uint32_t)MSG_RAM.RXF0) / 4);
+
+    SOC_CAN->SIDFC |= flssa << FDCAN_SIDFC_FLSSA_Pos;
+
+    SOC_CAN->RXF0C |= ((f0sa << FDCAN_RXF0C_F0SA_Pos)
+                       | (3 << FDCAN_RXF0C_F0S_Pos));
+    SOC_CAN->RXESC |= ((7 << FDCAN_RXESC_F1DS_Pos)
+                       | (7 << FDCAN_RXESC_F0DS_Pos));
+
+    SOC_CAN->TXBC |= ((tbsa << FDCAN_TXBC_TBSA_Pos)
+                       | (3 << FDCAN_TXBC_TFQS_Pos));
+    SOC_CAN->TXESC |= 7 << FDCAN_TXESC_TBDS_Pos;
```

Comment:
Out of curiosity, can these registers be set explicitly instead of using read-modify-write sequences?

Reply:
Yes, Modified to write register value directly.